models/demos/mamba/demo/demo.py

```diff
@@ -0,0 +1,152 @@
+# SPDX-FileCopyrightText: © 2023 Tenstorrent Inc.
+
+# SPDX-License-Identifier: Apache-2.0
+
+import time
+import argparse
+from typing import List
+from loguru import logger
+import ttnn
+import pytest
+import torch
+import json
+from tqdm import tqdm
+
+from transformers import AutoTokenizer
+
+from models.demos.mamba.reference.decode_model import MambaPretrainedModelName
+
+
+def get_cpu_reference_model(version: MambaPretrainedModelName, batch_size: int):
+    from models.demos.mamba.reference.decode_model import MambaDecode
+
+    return MambaDecode.from_pretrained(version, batch_size=batch_size)
+
+
+def get_tt_metal_model(version: MambaPretrainedModelName, device: ttnn.Device, use_cache: bool, batch_size: int):
+    from models.demos.mamba.tt.full_model import MambaTT
+    from models.demos.mamba.tt import model_config
+
+    reference_model = get_cpu_reference_model(version, batch_size=batch_size)
+    if use_cache:
+        cache_path = f"/tmp/{version}"
```

Comment:
is this the correct path for demo version? what are other models doing?

Reply:
Others models are setting path in model args. Set default to there model directory. Added a method to parameterize cache_dir.